### ISSUE TITLE
You need help with Graphics?

### ISSUE THREAD

**Issue by Methuselas** [alariq/mc2#5]
I used to work on Mechcommander Omnitech. I had issues, though, with the importing of animations, as I work mostly in Maya, not Max. If you can get me Bone support for animations and get rid of awful Targas, using something better for GPUs, I'll update all the graphics for you, since you're actually building an updated 64bit port for Windows/Linux. I actually have some models I built for the game, but never used. Problem is they ran about 5-6k polys which was too much for MCO to use.

Shoot me an email at methuselas@gmail.com if you'd like a graphic artist. :)

-M

**Comment by alariq** [comment]
Hi, Methuselas

This would be great. I also was thinkng about moving away from TGA's. You are right that high poly models kill performance in MC, because all T&L is done on CPU and in addition game is not multithreaded. 

Currently I at least want to finish all small things and put the game somewhere so everybody can try it and if there will be enough demand for this game then I can think about rewriting T&L and other part of the engine, because it is quite a big chunk of work, although not that complicated and it will simplify code at the end. After this bone support can be added, though it is also not a one day work.  Though I do not have much free time, so we'll see how it goes.

In any case, thank you for your interest!
I'll keep you updated if I be starting on any of those changes.

**Comment by Methuselas** [comment]
You should really go talk to Magic about the MCO build. It's only 32bit, but you're already further along with the engine rebuild, you two might be able to share code. His Mech Bay is amazing, as it allows for adding or reducing tonnage, hiring mechwarriors , doing repairs, etc. He stripped the engine of all the excess UI that was unneeded. I posted a link about your build on the forums. You can find them here:

https://www.hard-light.net/forums/index.php?board=210.0

As for animations, trust me, I've picked apart the ASE files over the years. It's not using actual bones, but helpers (or locators in maya) to move the polys. It's a bit of a pita and an antiquated system. I've literally pulled a model, converted it to an FBX in Max, imported it into Maya, deleted all the polygons, added my own, renamed them all to match the original polygon names, parented them to the corresponding joint helper, imported it back into Max, exported it as an ASE and it still crashes the engine. Only thing I can think of, is it's something to do with importing to Maya, but I don't use Max. I'm going to keep messing with it, since my clan died in MWO, so what little free time I have now, is back with MWO. 

Last message I read was that Magic had reached the limit of what he could do with the Engine, when I popped back in, so I started looking for a replacement engine. That's when I came across your build. I wanted to test it out, but my wife couldn't build it in VS (she's the software developer, not me. :P)

I really want to rebuild all the models and had planned to, but the limits with tgas, low poly count and no shader support had me migrating off to do other things. If it's not fun to do, you don't end up doing it. I got an itch to play MC2 again and then saw your build. A 64 bit MC2 engine would be outstanding.

**Comment by alariq** [comment]
I see, have not looked that much at animation code as math was not the main issue when porting the engine. I understand your concerns with animations now. 

Building code in VS should be quite straightforward if one has experience with CMake (it is just a matter of setting up paths to 3rdparties like SDL correctly), but I am planning to add some instructions to README file. I am also working on automated generation of all binary resources and packs from source data, so it is easy update game resources. It is currently in "working on my machine" state, but I already can play with my own generated resources and finished the game on Linux.

**Comment by Methuselas** [comment]
As soon as you can get me a way to build a Windows version, let me know and I can bug test for you, at least on the Windows side. I've been itching to play vanilla MC2, but I miss the video files, so I'll probably dig out my old disc. ;) I've got a bunch of FIT files and CSV files that are more true to tabletop I've been wanting to test out. 

I may just break down and watch some intro to max tutorials on Digital Tutors. When I left the Amiga scene, I toyed with Max in 2000, but the UI was so kludgy, I quit. Then, I took a couple of Maya courses and ran with Animation. Snagged me a few jobs in the game industry for a while.  Now that I'm a Dad, this is more for fun. My son likes watching me play MWO, so I showed him MCO and now he wants to play it.

**Comment by Methuselas** [comment]
Hey Alariq,

I've managed a workaround with Max and Maya. I successfully imported a mech, in game this weekend and passed a converted ASE animation to BVH (motionbuilder) back and forth between Max, Maya and the MC2 engine. I'm going to make a rig for Max and Maya, build generic animations to replace the current ones and update the game graphics. I want to do them in batches, so if you'd like copies of them, when I finish (I'm working on vehicles as well), let me know and I'll post you archives on my google drive. I'm working in vanilla MC2, since I need the Encyclopedia to view them.

**Comment by alariq** [comment]
That is great. Sure I'll be glad to test them. By the way I also ported Mechlopedia (in fact this was the first thing I did : -) )

**Comment by Methuselas** [comment]
Don't get your hopes up just yet. It was proof of concept. The bones I converted from helper bones still show up in the engine, instead of being invisible. They are moving, however, which is what I wanted. I'm still working the kinks out.

When testing, I jump back and forth between MCO and Vanilla. I use Vanilla's Encyclopedia to to test the fit files to make sure they're working and the mission editor to check the models. MCO supports up to 10k polys now (I wasn't aware of the update) and 512k targas, which is my test bed for graphics. 

BTW, concerning the Encyclopedia, are you going to remove the string.txt from the dll file? I know you can just use 7zip to remove it for editing, but an external xml or an xls file would be so much nicer to use.

**Comment by alariq** [comment]
you mean string.res.cpp/.h. Yes I was thinking about it. But at the beginning for me as a programmer it was much easier to write a simple script which would grab all strings and embed them right in cpp file )

**Comment by alariq** [comment]
for the strings. Easy way (if you have compiler) is to just add/modify sttring in text file and recompile dll.
AFAIR, indeed, each mech has only one texture, but even if there would be ability to hide bones by rendering them with "null" material, it does not sound like a good decision. In fact all this part has to be rewritten.

**Comment by Methuselas** [comment]
Yeah, I see your point. I'm always looking at things from a modding perspective. The more a game has an ability to mod, the more likely you are to attract new developers. A lot of people can't compile and truthfully, I never understood GitHub enough to add commits to the other game project I'm working on.

So, are you planning on re-writing all the animation routines? I mean, I can deal with the animations issue, now that I know what I was doing wrong this entire time. I even setup a VM workstation test bed, just so I could install an old copy of 3DS Max 4.2 to see if I could maximize the limitations in the animation code. If you are going to do this, I would suggest keeping legacy code, to do it the old way, until, at the very least, I get new skeletons and animations done for all the different mechs. (I believe there's only 4 or 5 different helper skeletons in the game.)

Let me know when you're ready and I'll build you a rigged mech to test in game. I do have a simple request, though, if you have the time. Can you have the engine look for "variant" mech models in the mech cvs? Basically, have a "mechname" under the variant section that points to a different model, with another weapon loadout. If it doesn't fine one, it automatically uses the default mech model.

**Comment by alariq** [comment]
I understand "modding viewpoint" as well. It is just that too much time needed to make everything "the right way". So for me it was a shortcut to write script to autoconvert all strings and embed them in the code. Though compiling is not that scary :) in most cases it is just a matter of one button press. Way too easier than making an anmimation.

I do not think that I'll be able to change animation code in near future. Task number one for me (after sorting out all infrastructure issues, so that anyone can easilly build code and data and run game) - is to rewrite model loading and rendering  code, which is quite a big piece of work, but it will make everything look simpler and will allow to strip away a lot of code. From more high level perspective it will remove hard constrain on the amount of vertices/textures in the models and allow for more sophisticated lighting/shading as it will be performed on GPU, not to mention performance boost.

Though, if there will be no interest to the game I might as well loose motivation )

**Comment by Methuselas** [comment]
Don't lose motivation! :( I've already started doing new models for kit bashing. Besides, you're the closest I'll get to being able to make a setup as close to tabletop, as possible. Also, is the HBSTATZ.csv, where we would find all the stats for ranges for weapons?

I mentioned your build on the old MCO site. I was going to mention it on the MWO forums, but I wasn't sure if you wanted that many inquiries so soon. Like I said, I stumbled upon you accidentally.

**Comment by alariq** [comment]
I can't find any reference on HBSTATZ.csv in the code/data. Not sure if it is needed

**Comment by Methuselas** [comment]
I think that's for range values, as well as a few other in-game things, but I'm not sure. BTW, I hope you're still motivated, as I have this completed. I haven't UV mapped it, but it's 10k tris, which is about 10 times what we got in the original game and it's half the size of the MWO model I based it on. ;) 

[https://i.imgur.com/wrmNNYg.png](url)

**Comment by alariq** [comment]
Nice! Though, I think MC2 will choke tryint to crunch all the vertices ) . Until I'll move T&L to GPU, this, probably, will not fly, but we can check.

**Comment by Methuselas** [comment]
I know Magic gave me a polycount of 3-5k in MCO, which worked, but you DID say you wanted 2017 graphics. ;) Believe it or not, this is the low poly mode, I prepped to do a high poly version for normal maps I planned on baking into the diffuse textures. I could probably drop it to 7k, by removing some of the raised edges. It's missing some of the extra polys from the original model, which makes me sad, but this is done in pieces so I can kit bash other mechs that are missing.  

I'll build you a test mech with all the animations if you want to check memory usage.

**Comment by Methuselas** [comment]
Okay, it may be a while before I get you a working rig. I got the thing UV'ed to a single map, but once again, I'm breaking the ASE files for some reason. Even exporting a source ASE file, with no modifications is still crashing the game. It's one of the things that frustrated me with the MCO mod group, as I could never get a proper walk-through on how to export the models in ASE format for the game. I'll still keep building high poly models, though. They're just additional portfolio pieces.

**Comment by alariq** [comment]
you may send me you ASE example and I might be able to figure out why it crashes the game.
I know at least one bug I fixed when loading ASEs

**Comment by Methuselas** [comment]
https://drive.google.com/open?id=0BwpB_jpXgx3Ia25ZZ2ZxRExwTTQ

Here's a link for two. One is a Catapult, that I linked myself, using a helper skeleton. The second is an Anubis that I exported from Max 7, as a Max file and imported it into 2017. I then exported it back out as an ascii file. Both crashed the Encyclopedia and the Editor, neither loading. Ironically, the helper skeletons work fine in Max.

**Comment by Methuselas** [comment]
Merry Christmas!

Here's the Flea/Fireant

https://i.imgur.com/T7nXVSJ.png

Alariq,

Do you know from the code how many LODs MC2 supports? I know, it's at least one.

**Comment by alariq** [comment]
The problem with your file is that it does not have normals info in it (as well as texture UVs). Loading code in MC2 is quite simple, it does not check for any of those cases and just crashes it this happens :-)

**Comment by Methuselas** [comment]
Yeah, it's a bit more than that. I figured out the normals issue after reading the ASCII code by hand. Also, if the helper objects aren't set to "bones" for the skeleton and the weapon helpers set to "point", it won't work properly. It really doesn't care about UV info, which is funny. I'll write up a tutorial to add mechs to the game, when I work out all the kinks with using the original animations (the Catapult test above shifts in elevation. I need to find out just how many skeletons are used by the engine, so I can redo all the animations. Making generic animations for generic skeletons is the easy part.

**Comment by Methuselas** [comment]
Is there any chance you could update the readme to cover compiling on Windows? The developer of Mechcommander Omnitech has taken an interest in your build. He might be contacting you about code sharing.

**Comment by alariq** [comment]
Sure, it is very easy. I'll add some notes

**Comment by Methuselas** [comment]
[https://drive.google.com/open?id=0BwpB_jpXgx3Ia25ZZ2ZxRExwTTQ](url)

Not sure if you want to set these up, but here's all the movies ripped off the original MC2 CD. If you decide to add in the support for bik files, so these will play, I might be persuaded to redo these as CGI elements so they can be fully GPL. They honestly add to the game. It's a shame that Microsoft left them out.

**Comment by Methuselas** [comment]
Hey, Alriq, can you take a look at this. I had it working properly, no problems. Then, I went to optimize the UV maps, overlaying and flipping UVs to maximize UV space and then out of nowhere, I started seeing transparency through the mesh. I've literally done everything from remapping UVs, duplicating, cloning, deleting history, etc, to no avail. The weird thing is, it seems to have corrupted all my previous saves, as well. The only one that works was the original model, before I updated the model, optimizing the triangles, that I exported as an FBX to paint. Of course, when I change it from tris to quads, by hand, it does the exact, same thing as my other models. Even my Flea and Wolfhound models are doing this (This is the Locust, saved as the Fireant.) 

https://drive.google.com/open?id=1a-UeBvfo-1gF58qLWxjyq-NacBXShenO

If I can figure out why this thing died on me (I wondered if it's due to the UV stacking and flipping, but I've tried redoing the UVs to no avail), I can package it for you, with 2 LODs and maps for diffuse, spec and normal to test the GL shaders.

**Comment by SeanLang** [comment]
@alariq do you have a current version of this that runs on win 10 64bit? I can not get the source or omnitech to run.

Also what are your goals for said project?

**Comment by alariq** [comment]
@SeanLang  yes windows version runs on win 10 64 bit. you need to build it (ther is a read me instructions) and also build data package from other repository (mc2srcdata)

@Methuselas glad you resolved your issue, wanted to look at it, but was distracted.

**Comment by Methuselas** [comment]
https://i.imgur.com/mrUj1vE.jpg

Didn't really get it resolved. All of the sudden, my mats no longer work. My workaround is doing all the work in maya, sending it to max, exporting as an obj and then re-importing it back into Max to do the final editing (which has to be done in Max anyway). Otherwise, I get some strange backface culling that I can't figure out. It's a PITA, but as you can see in the above image, even a simple poly cube does it. Not sure why or what happened. I even did a complete reformat thinking it was some sort of corruption issue, as Maya says there's something wrong with my files, but there are no errors in the script editor.

**Comment by SeanLang** [comment]
@alariq unfortunately I'm not familiar with visual studio/building projects. Any chance you could do a build for win10 64bit? If not I understand!

**Comment by alariq** [comment]
@SeanLang I'll try to do it next week. I've got new win PC so it will be a good experience to build everything from scratch and ensure it all works

**Comment by alariq** [comment]
@Methuselas maybe yoiur triangle winding order is not consistenst across a model? CCW vs. CW?

**Comment by Methuselas** [comment]
No, it's some sort of bug. I have a work around, which is just a single, extra step and I'm brushing up on 3ds Max to see if maybe I can figure out if it's a material problem. Like I said, if I export the model, as an obj, even the original material parameters work fine. As soon as I send it back to Maya or import it into Maya as an obj, I'm missing polygons. 

This did, however, work in my favor. As I thought my original files were corrupted, I deleted them and started over, completely. I rebuilt the Locust and shaved off about 1300 tris, so it's sitting at just over 11k, instead of 12.5k. I can reduce it even more in testing, using Poly Cruncher, but not by much, if I want to keep my edges.

**Comment by Methuselas** [comment]
One more thing, if you want me to build a test model for materials, let me know. I can make a Locust for you, labeled as the Fireant, with normal, spec, diffuse and AO maps. I have it all UV'ed, I just haven't optimized them to maximize texture space, yet.

**Comment by SeanLang** [comment]
@alariq awesome, I'd love to play this again!

**Comment by djzort** [comment]
A kickstarter campaign might be in order - to fund some dedicated hours of working through issues?

**Comment by alariq** [comment]
@Methuselas : currently even in my branch which uses hardware vertex transform I do not support lighting, I may hack it though. But I'd wait a bit untill I'll gather and pass all lighting info in pixel shaders. This will be a good model to test

SeanLang: Haven't tested it on 2560x1440 (only up to full hd), but I do not see why it should not work. The only poroblem right now would be to correctly position all UI elements.
Kickstarter campaign sound aewsome, but I am not sure that there will be enough interested people. And the game is not tested enough.

**Comment by Methuselas** [comment]
I'll do the Urbanmech, instead. I just started modeling it and it should be able to use the original animations, without my having to make new ones, as I've been intending. I need to do animations for the Locust and Fireant. They don't play well with the original animations. I haven't even bothered yet with the Commando, Firestarter, Wolfhound or Raven. They're all not uv'ed yet. I've been focusing on getting models done to kitbash the ones that are missing from MC2.

**Comment by alariq** [comment]
@Methuselas I've updated README.txt so it should be easy (ok, not hard )) to build code + data.
Sorry, it, probable have taken you some time to figure out all steps to buid the game. But how did you went with data files? I am using different font file format and only recently uploaded scripts for generateng them )

**Comment by Methuselas** [comment]
LOL! Simple. When I borked the code, I asked my Linux guru coder friend to fix my compile. XD

**Comment by alariq** [comment]
I see, I hope, with updated README it will be simple enough for a person who does not have a Linux guru friend ))

**Comment by alariq** [comment]
Him @Methuselas I know I am not very active here, but I've got some more time lately to work on shader based lighting, first I'd like to make existing assets work and then we can test your model. Or you can prepare this one to be loaded by existing code (I mean export it and add necesary agl file, etc.), so I can test with it.

**Comment by Methuselas** [comment]
> 
> 
> Him @Methuselas I know I am not very active here, but I've got some more time lately to work on shader based lighting, first I'd like to make existing assets work and then we can test your model. Or you can prepare this one to be loaded by existing code (I mean export it and add necesary agl file, etc.), so I can test with it.


So far, none of my models work well, with the stock 3DMax skeletons. I'm seeing a lot of artifacting. Best for now, to work on what you're doing, before we change all the assets.

Also, I noticed with the latest commit, we have fog in shaders now? Does this mean T&L are shader based?

**Comment by alariq** [comment]
> 
> 
> > Him @Methuselas I know I am not very active here, but I've got some more time lately to work on shader based lighting, first I'd like to make existing assets work and then we can test your model. Or you can prepare this one to be loaded by existing code (I mean export it and add necesary agl file, etc.), so I can test with it.
> 
> So far, none of my models work well, with the stock 3DMax skeletons. I'm seeing a lot of artifacting. Best for now, to work on what you're doing, before we change all the assets.
> 
> Also, I noticed with the latest commit, we have fog in shaders now? Does this mean T&L are shader based?

Currently everything is shader based, but only solid models ( mechs, some building etc.) are lit in shaders and drawn not as bunch of triangles pretransformed on CPU but as BV + IB transformed and lit in shaders

**Comment by Methuselas** [comment]
Alariq, if you find the time to update, I've got all the mech animations exported from ASE to FBX. Apparently Z is up in MC2, so I'll have to fix them, since I use Y-Up.

But I did get a Mechcommander Online Battlemaster rigged to the MC2 Atlas skeleton. ;) Here's the repaired walkcycle for the Atlas, using the Battlemaster.

https://drive.google.com/file/d/1fiH24H8jkrH-rvdKy1HLRuepAGK-Ntv3/view?usp=sharing

If you can get me FBX and PNG support for models and textures, I'll start updating the original artwork. I use the MechcommanderOmnitech mod, since it allows me to not have to use PAK files and gives me 256x256 texture maps to test and I don't have a windows build of your version. I couldn't compile it and couldn't find someone I could hire that was willing to build it, who wasn't trying to gouge me. -_-

Hope all is well with you.

**Comment by Methuselas** [comment]
Oh and Alariq, I got all the animations and models ripped from the Battletech Game by HBS. ;) I gotta do cleanup on them all, but if you want me to make you some ASE files of a mech or two, so you can test shaders and animations, let me know.

**Comment by Methuselas** [comment]
Alariq,

Just an update. I've cleaned up the rigs for the default MC2 Mechs up to "R." I'm missing one: the Men Shen. There's no good model available, which means kitbashing one. Vehicles are another story. The majority don't have models, so I've downloaded STL files to remodel for MC2GL. Textures are also an issue. A lot have really bad ones, especially broken normals. I made a quick and dirty PBR material for the Atlas, at 2k, so you can test shaders, but it's going to cost you in Gfx memory, using TGAs. Another thing on the Atlas shader - I'm pretty sure I burned the normals in DX11, so when I send them to you, let me know if they don't work right, as I'll have to re-burn them to GL. 

Also, these models aren't optimized for MC2, so they're pretty poly heavy - between 20k-70k for just the models, without weapons. I don't really want to reduce the polycount, TBH, as I have to do it by hand, to keep the UVs together. I do have some tools that may help, but it's not a priority right now. I'm going to install 3DMax this weekend, to export the mechs to ASE. You should be able do just dump the ASE files in the TGL folder. MC2 automatically converts ASE to tgl it finds, as soon as the game starts loading. 

As for animations, Battletech came with 64 anims and all but one can be reused for MC2GL. The missing one is GetUpFront, but I can repurpose one of the falling anims, by reversing it. Now, each mech comes with an ini file. Inside that file are elements for Gestures. Can you get me the corresponding name of each gesture to the anim file. I've never been able to find them, nor do I know if the anims need to match case to the originals, due to hard coding. 

I can only test these models and anims using Mechcommander Omnitech. If you could get me a Windows executable, I'd really appreciate it.  I can't compile it. Coding isn't my strong suit.

Hope all is well.

**Comment by alariq** [comment]
Hi, @Methuselas. It is great that you have that much interest, I will have to look for those gestures and maybe can provide a Windows build, but it will not be very soon. as for normal textures, shaders have to be updated for those, this will require some time as well, because tangents have to be calculated and passed along with vertices, nothing hard but requires time and I am devoted to some other things at the moment. I'd like to return to it some day but not sure when.

**Comment by Methuselas** [comment]
It's no worries. I doing this for a series of animated shorts I'm plotting, about a merc company, circa 3025. I'm converting these all over to iClone, painting them 8k textures and then using them as the character's mechs. Since I'm doing all that work, since I already will have them and I actually loved this game (it's my favorite of all the BT games), I see no reason to not offer them up to you for this port. :) MCO is getting harder to run, it's still 32bit, doesn't offload on the GFX card, nor does it have shader support. 

If  I could get a 64Bit version with an editor that wasn't hard-coded, I'd be a happy man. 

I'm actually about to re-install 3DMax, so I can test this. I know it works, I've done it before, but with smaller poly versions. Once I see an Atlas walking in the ViewRel and in the Editor, I'll 7z it up for you and shoot it to you, along with the PBR textures I did, do you can do debug easier. I'm going to be busy for a while anyway. I still haven't figured out how to set them up as characters in iClone yet, so it's easier to kitbash and make variants. Not to mention, I have over 300 mechs to convert, as well as around 100 vehicles from all Eras up to the Dark Age. I don't need them all....

.... but I want them. :D

BTW, I actually own the game and if MS just disabled the BIK code instead of removing it, let me know if you want all the original movies for Carver V, as well as the Pilots. I think I have them saved somewhere already on an external drive.

**Comment by Methuselas** [comment]
[https://drive.google.com/file/d/1XsKdxjM0jcyTE1vnMx4fvYTKVEWYYDwe/view?usp=sharing](url)

Alariq, 

See if you can load this. the ASEConv is crashing on mine. I think it has something to do with it trying to use my intel chip, instead of my GeForce. I've never seen it do this. The TGA is 256, though. I was trying to open this in MCO and it didn't work, so I used the Final Build's ASEConv with the Source, but it was still no joy. At first, I thought it might be the Skeleton, as I've never actually used a non-MC2 skeleton, so I put this on the MC2 Atlas skeleton and it still wouldn't load for me.

**Comment by Methuselas** [comment]
Disregard that Atlas I just sent you. I completely forgot that MC2 doesn't recognize bones, only helper objects. I thought it might, since the skeletons were made with CAT, but MC2 was made using 3DSMax 3 and doesn't recognize them. I also forgot that the meshes have to be linked in 3DMax, in a specific way to the helpers or they won't work, so that Atlas isn't going to work, at all. I am going to have to go back to the drawing board on replacing mechs, using the original skeletons and animations. Thankfully, I still have the old MAX skeleton that Magic gave me, when I was working on MCO. I just have to remember the steps to do it. In my genius, I forgot to write the steps down. 🙄

In other news, while looking for notes, to figure out why it wasn't working, I found my old project folder that had a few mechs that were reduced in polycount by about 90%. Instead of 20k-50k, they're 2k-5k. They're all Light Mechs, though and only one or two are actually from the original MC2. They're also don't have UVs. I think I stopped working on these, when Magic said he didn't want to add them to MCO, as he didn't like the MWO-style Battlemechs and they were still too heavy in poly count for MCO. I honestly can't remember, as it was 8 or 9 years ago.

**Comment by Methuselas** [comment]
[https://drive.google.com/file/d/1o6zFkMh41d-YU_YnG2YHxyjkr1JrmVhL/view?usp=sharing](url)

Try this one. I got it to load in the Editor and it took about 10 minutes to load :P, but it appeared to load fine. It will use the default Atlas animations. I managed to load it with the ViewRel, but during the walk anim, the legs and one foot disappeared. XD There's no textures.

I don't think I'll be able to convert the other animations for you, but I can rig new models to the old animations, which really aren't that bad. I'll make a texture tomorrow and rebuild the Atlas.

**Comment by Methuselas** [comment]
[https://drive.google.com/file/d/152zKL9kymiOBEMoYcBNuNqxqUIgyeI8z/view?usp=sharing](url)

Okay, I remember how to do it now. Here's my re-poly of the MWO Raven, using the Original Raven animations. No textures.

**Comment by Methuselas** [comment]
[https://drive.google.com/file/d/1cFOdSor7i0vuIMvbX8gwogkWK7MGRy-U/view?usp=sharing](url) 

BT Atlas rigged to original Atlas skeleton from MC2. With Textures. There's a 512k TGA for loading normally and 2k PBR textures I did, for your shader tests. MCO has a 10k model limit, which is why the legs weren't appearing in the ViewRel (and why I wanted a windows build of your version :P). This should be enough to get you started to work on shaders and high-poly models. Now that I know the steps to do it correctly again, I'll convert more mechs and vehicles, if you choose to go this direction.

Now that I know where you are, stay safe. Nothing more dangerous than Uruk-hai. ;)

**Comment by Methuselas** [comment]
Alariq,

Off topic, but I've figured out how to make MC2 use skeletons, so I can build mech rigs. It just requires me do all the animations in Maya and export the base figure and the animations as FBX. Import it into Max, set the bones to convert to dummies and volia! No more detached dummy bones that don't move correctly. :) Export to ASE and we're done. :)
